src/steps/data_validation/utils/dataset_validator.py

```diff
@@ -0,0 +1,145 @@
+import os
+
+from PIL import Image
+
+from src.models.dataset.dataset_collection import DatasetCollection
+from src.models.dataset.dataset_context import DatasetContext
+
+
+def get_image_path_list(image_dir: str) -> list[str]:
```

Comment:
The return type should be `List[str]`, otherwise we would lose backward compatibility with Python 3.8 and 3.9